[WIP] Extract remaining rake task bodies

backus opened the pull request:
Continuing in-place refactoring of rake tasks.

mbj commented on lib/devtools/rake/mutant.rb:
```diff
@@ -0,0 +1,63 @@
+    class Mutant
+      include Procto.call(:verify),
+              Adamantium,
+              Anima.new(
```
Why dont you simply pass the `Devtools::Config::Mutant` object into the `Devtools::Rake::Mutant` one? Reducing the need to duplicate the attribute definition, both here and in any call side?

mbj commented on lib/devtools/rake/reek.rb:
```diff
@@ -0,0 +1,24 @@
+  module Rake
+    # Reek metric runner
+    class Reek
+      include Procto.call(:verify), Adamantium, Anima.new(:files, :config_file)
```
Instance of: https://github.com/mbj/devtools/pull/112/files#r43601041

mbj commented on lib/devtools/rake/rubocop.rb:
```diff
@@ -0,0 +1,25 @@
+  module Rake
+    # Rubocop metric runner
+    class Rubocop
+      include Anima.new(:config_file, :directory), Procto.call(:verify)
```
Instance of https://github.com/mbj/devtools/pull/112/files#r43601041

mbj commented on lib/devtools/rake/rubocop.rb:
```diff
@@ -0,0 +1,25 @@
+
+      # Verify rubocop integration runs successfully
+      #
+      # rubocop:disable Style/RedundantBegin
```
You actually should IMO lift the rescue to def level, killing the begin/end that is redundant.

backus replied:
I agree. I decided to extract before refactoring since that is where I am most likely to lose momentum

backus replied:
Still working on extracting flog

backus commented:
@mbj I changed class bodies to use `config` instead.  This was a quick change and I've committed the changes as `fixup!` commits so you can see the changes by themselves. I haven't done the refactoring step yet but if you would like to do another pass on the extraction itself I've addressed your flag

mbj commented:
@backus I like the direction this is going. In case you do not have the time to finish it, I'll take this PR over.

backus commented:
@mbj Go for it. I can still contribute some work on weekends.  If you start paving out a general pattern for how you refactor the different tasks I can follow and apply that pattern for others

mbj commented:
> @mbj Go for it. I can still contribute some work on weekends. If you start paving out a general pattern for how you refactor the different tasks I can follow and apply that pattern for others

I'll focus on one specific tool (mutant). Once we are happy with that one we can introduce the same thing for the others.

backus commented:
> I'll focus on one specific tool (mutant). 

Great.  Note https://github.com/mbj/devtools/commit/018a41321623a528c90d55eca22cc79f54387940 in that case. I had to change the task body a bit before adding tests become

```ruby
namespace =
  if config.zombify
    Mutant.zombify
    Zombie::Mutant
  else
    Mutant
  end
```

isn't valid given the mutant public API anymore

mbj commented:
> isn't valid given the mutant public API anymore

Yeah, I plan to go through the CLI API.

backus commented:
I still do have a separate branch locally which captures Flay's reporting as a string instead of having it directly report to STDOUT.  I'll try to pull that up as a separate feature branch later
